flight/UAVObjects/uavobjectmanager.c

```diff
@@ -2177,6 +2182,42 @@ void UAVObjRegisterNewInstanceCB(new_uavo_instance_cb_t callback)
 {
 	newUavObjInstanceCB = callback;
 }
+
+static int32_t cleanupSettingsObjs(void)
+{
+	extern uint32_t __uavobj_settings_register_start[], __uavobj_settings_register_end[];
+	uint32_t slot = 1;
+	uint32_t curr_obj_id, curr_obj_inst_id;
+	int32_t res = 0;
+	do {
+		res = PIOS_FLASHFS_ObjIdAtSlot(pios_uavo_settings_fs_id, slot++, 
+					&curr_obj_id, &curr_obj_inst_id) == 0;
+
+		if (res == -3) {
+			// empty slot, ignore it
+			res = 0;
+			continue;
+		}
+		
```

Comment:
We should handle other error conditions here (`res` can be negative and the code below gets executed anyways).